esphome-dev/rootfs/etc/cont-init.d/30-esphome-fork.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/with-contenv bashio
+# ==============================================================================
+# Community Hass.io Add-ons: ESPHome
+# This files installs the user ESPHome fork if specified
+# The fork must be up to date with the latest ESPHome dev branch
+# and have no conflicts
+# ==============================================================================
+
+declare esphome_fork
+
+if bashio::config.has_value 'esphome_fork'; then
+    esphome_fork=$(bashio::config 'esphome_fork')
+    if [[ $esphome_fork == *":"* ]]; then
+      IFS=':' read -r -a array <<< "$esphome_fork"
+      username=${array[0]}
+      ref=${array[1]}
+    else
+      username="esphome"
+      ref=$esphome_fork
+    fi
+    full_url="https://github.com/${username}/esphome/archive/${ref}.zip"
+    bashio::log.info "Checking forked ESPHome"
+    dev_version=$(python3 -c "from esphome.const import __version__; print(__version__)")
+    bashio::log.info "Installing esphome from fork '${esphome_fork}' (${full_url})..."
+    pip3 install -U --no-cache-dir "${full_url}" -qq \
+      || bashio::exit.nok "Failed installing esphome pinned version."
+    fork_version=$(python3 -c "from esphome.const import __version__; print(__version__)")
+
+    if [[ "$fork_version" != "$dev_version" ]]; then
+      pip3 install -U --no-cache-dir https://github.com/esphome/esphome/archive/dev.zip -qq
```

Comment:
Does `bashio::exit.nok` abort container startup?
If so, reinstalling the dev version shouldn't be necessary.
If not, it probably should as the config value is invalid.

Reply:
Yes it aborts the startup. I am not sure about how supervisor starts containers (whether it recreates them or just starts them again) and so I did it like this because if it failed due to version, you could remove the version config and it would start using the fork

Reply:
The supervisor stops and removes containers. They are recreated on their next start (from the images downloaded or created during addon installation), so the fork installation will never be persistent as long as you don't install it to the /ssl or /config folders mapped into the container.

Reply:
I have just confirmed this too. Thanks for pointing it out